### DISCUSSION TITLE
Postfix Increment

### DISCUSSION THREAD

**Discussion by halitaydin** [eslint/eslint#14237] [Q&A] [answered]
Is there any postfix increment option in Eslint.

Eslint advice me to use '+=' instead of using ++.
If I use '+=' the result changing. 

Please find the examples below.

(Without eslint rules. Result = 1, 2, 3, 4, 5). _(my expectation)_
<img width="556" alt="without Eslint rules" src="https://user-images.githubusercontent.com/64103256/111905174-6c62bf80-8a5b-11eb-985b-e54e7feb8b2c.png">

(With eslint rules. Result = 2, 3, 4, 5, 6).
<img width="564" alt="with Eslint rules" src="https://user-images.githubusercontent.com/64103256/111905184-75ec2780-8a5b-11eb-8c68-ba26bd82e253.png">

Note: I have faced this problem when working javascript on javascript.info website.

**Comment by mdjermanovic** [comment, marked as answer]
Hi @halitaydin!

> Eslint advice me to use '+=' instead of using ++.

Can you please share the error message you're getting?

**Reply by halitaydin** [reply]
<img width="556" alt="without Eslint rules" src="https://user-images.githubusercontent.com/64103256/111953414-a2558180-8af7-11eb-872f-7ccb408dc534.png">

If I use 'this.current++' eslint warn me with this message:
Unary operator '++' used.

And if i use '+=' according to eslint rules, the result changing.

Thank you for your help.

**Reply by mdjermanovic** [reply]
Thanks for the info! The error was reported by the `no-plusplus` rule.

> And if i use '+=' according to eslint rules, the result changing.

The rule doesn't suggest replacing `++` with `+=` at the same place. It just disallows the use of `++` and `--`, while the fix depends on the particular code.

In your example, the code can be rewritten to:

```js
const result = { done: false, value: this.current };
this.current += 1;
```

If you prefer incrementing `this.current` in the same expression where its value is used, then I don't think there is really a "better" alternative to `++`, so you can disable this rule:

```json
rules: {
    "no-plusplus": "off"
}
```

This is mostly a stylistic rule.

**Reply by manavm1990** [reply]
`    "no-plusplus": ["error", { allowForLoopAfterthoughts: true }],`

**Reply by halitaydin** [reply]
> Thanks for the info! The error was reported by the `no-plusplus` rule.
> 
> > And if i use '+=' according to eslint rules, the result changing.
> 
> The rule doesn't suggest replacing `++` with `+=` at the same place. It just disallows the use of `++` and `--`, while the fix depends on the particular code.
> 
> In your example, the code can be rewritten to:
> 
> ```js
> const result = { done: false, value: this.current };
> this.current += 1;
> ```
> 
> If you prefer incrementing `this.current` in the same expression where its value is used, then I don't think there is really a "better" alternative to `++`, so you can disable this rule:
> 
> ```json
> rules: {
>     "no-plusplus": "off"
> }
> ```
> 
> This is mostly a stylistic rule.

```javascript
const result = { done: false, value: this.current };
this.current += 1;
```

That's it. Thank you for your help @mdjermanovic .
